source/common/upstream/eds.cc

```diff
@@ -118,10 +119,34 @@ void EdsClusterImpl::onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt
                                      cluster_load_assignment.cluster_name()));
   }
 
+  // Reset assignment timeout (if set) as we have received new assignment.
```

Comment:
Do we also want stats for this? See other places in this file where stats are used, e.g. `info_->stats().update_empty_.inc();`

Reply:
Addded two stats
 - for #assignment_stale information is received 
 - for #assignment_stale is triggred